sale_report_qweb_decimal_place/reports/sale_report.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- pylint:disable=deprecated-data-xml-node -->
+<odoo>
+    <template
+        id="report_saleorder_document_inherit"
+        inherit_id="sale.report_saleorder_document"
+    >
+        <xpath expr="//td[@name='td_priceunit']/span" position="replace">
+            <t t-set="currency" t-value="doc.currency_id" />
+            <t t-set="price_unit" t-value="line.price_unit" />
+            <t t-call="report_qweb_decimal_place.price_unit_value_format" />
+        </xpath>
```

Comment:
```suggestion
        <xpath expr="//span[@t-field='line.price_unit']" position="attributes">
            <attribute name="t-att-style">'display: none' if doc.currency_id.apply_price_decimal_place else ''</attribute>
        </xpath>
        <xpath expr="//span[@t-field='line.price_unit']" position="after">
            <t t-if="not doc.currency_id.apply_price_decimal_place">
                <t t-set="currency" t-value="doc.currency_id" />
                <t t-set="price_unit" t-value="line.price_unit" />
                <t t-call="report_qweb_decimal_place.price_unit_value_format" />
            </t>
        </xpath>
```
To avoid the overuse of `replace`.

I was wondering if we should actually change the following template from `report_qweb_decimal_place`
```
    <template id="price_unit_value_format">
        <t t-if="currency.apply_price_decimal_place">
            <span
                class="text-nowrap"
                t-esc="price_unit"
                t-options='{"widget": "float", "precision": currency.price_decimal_places}'
            />
        </t>
        <t t-else="">
            <span class="text-nowrap" t-field="line.price_unit" />
        </t>
    </template>
```
to
```
    <template id="price_unit_value_format">
        <span
            class="text-nowrap"
            t-esc="price_unit"
            t-options='{"widget": "float", "precision": currency.price_decimal_places}'
        />
    </template>
```
assuming that the template is called in the above suggested manner.  What do you think?

Reply:
Yes. I think this approach is better.

Reply:
OK, please go ahead and apply changes to all the relevant places.